src/auth/auth.service.ts

```
@@ -0,0 +1,305 @@
+    const decoded = this.jwtService.decode(token);
+    const needsRoleSelection = !decoded.role;
+    return { token, needsRoleSelection };
+  }
+
+  async validateUserByGoogle(profile: any): Promise<string> {
+    try {
+      
+      console.log('Raw Google profile:', JSON.stringify(profile, null, 2));
+      
+      const googleId = profile.id || 
+                      (profile._json && profile._json.sub) || 
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)